### ISSUE TITLE
Is it possible to define project makefile name?

### ISSUE THREAD

**Issue by JohannesMP** [premake/premake-core#211]
When working with a `gmake` solution, if `location` is set at the solution scope, the resulting files will look like this (assuming two projects):

    <ROOT>
        - SOLUTION_LOCATION_DIR
            - Makefile
            - PROJECT1_NAME.make
            - PROJECT2_NAME.make

with this behavior, I like that the name of the project (defined with the `project` function) is used as the make file's name.

However if I define location at the solution scope, and then also at each project scope, the files will look like:

    ROOT
        - SOLUTION_LOCATION_DIR
            - Makefile
        - PROJECT1_LOCATION_DIR
            - Makefile
        - PROJECT2_LOCATION_DIR
            - Makefile

Where, since there is no name ambiguity, the make files use the default naming convention. However I'd like to have the option of nonetheless having the name of the project make files correspond to the name of their project.

Is there a way to force the basic naming behavior, despite there not being any naming ambiguity, or at least manually define the make file name?

--------

So a premake5.lua file containing the following:

    solution "test_solution"
        location "."
    project "test_project"
        location "make"

currently creates the file structure:

    - Makefile
    - make
        - Makefile

and I'm wondering how I can make it be this instead:

    - Makefile
    - make
        - test_project.make

**Comment by starkos** [comment]
No, that is not possible currently. You would need to modify `make.getfilename()` in `src/actions/make/_make.lua` to make that change.

Ideally, we could look at the `this.filename` field, which is set by the `filename()` project API. But I'm not sure how we would tell if it had been set by the user via script (in which case we should use it) or automatically (in which case we should use the current behavior instead).
